### ISSUE TITLE
Mark character icons for review in color-contrast rule

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#946]
Character icons like "?", "!", "X" should be flagged for review when they don't meet the contrast ratio, rather than outright marking them as in error.

**Comment by marcysutton** [comment]
How would this work with other character sets...do those ever get substituted for other characters? Are there others we should include in the list?

**Comment by marcysutton** [comment]
@WilcoFiers I should have asked this before, but can you explain the rationale as to why these icons would go to review instead of a fail?

**Comment by WilcoFiers** [comment]
Quite simply, because they're not violations of WCAG 2.0. WCAG's definition of text excludes things like icons. They are considered non-text content. Now because we can't be certain about if "X" means "close" or if it actually means the letter "x", we shouldn't outright pass it, but we can't say for sure it's a failure when that character doesn't have the required contrast.

That's not to say there are no contrast requirements for that "X" meaning "close". WCAG 2.1 has a requirement for contrast of non-text content. It should be at least 3:1. That should probably be its own rule though. We don't want to mix WCAG 2.0 and 2.1 requirements into a single rule, because otherwise people only interested in 2.0 can't use it.

**Comment by dylanb** [comment]
@WilcoFiers Text that is presented as text (semantically) needs to pass the color contrast requirement. The problem here is not with the axe-core rules (unless we are not handling the case below), it is that the developers have not used semantic markup.

How do we currently handle it if the single character is given a role="img"? IMO, that is the way for the developers to eliminate this issue.

**Comment by WilcoFiers** [comment]
Closing it as a duplicate of #1068. I am pushing ahead with putting single character text elements into review. We have a bunch of separate issues open that are partially resolved by doing this. We can revisit this in the future, come up with something more robust.
